components/event-simulator/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/ds/EventSimulatorValueHolder.java

```diff
@@ -71,8 +73,11 @@ public static EventManagementService getEventManagementService() {
         return eventManagerService;
     }
 
-    public static Map<String, EventStreamProducer> getEventProducerMap() {
-        return eventProducerMap;
+    public static Map<String, EventStreamProducer> getEventProducerMap(int tenantID) {
+        return eventProducerMap.get(tenantID);
     }
 
+    public static void setEventProducerMap(int tenantID) {
```

Comment:
@ramindu90 Since this method is simply creating a new map and not setting the map, shouldn't the method name be something like 'createEventProducerMapForTenant(int tenantID)' or similar? WDYT?